clang/test/SemaCXX/cxx2c-binding-pack-nontemplate.cpp

```diff
@@ -3,9 +3,15 @@
 // RUN: %clang_cc1 -std=c++23 -verify=cxx23,nontemplate -fsyntax-only -Wc++26-extensions %s
 
 void decompose_array() {
-  int arr[4] = {1, 2, 3, 6};
```

Comment:
We should have both cases tested, we want as wide test coverage as possible. 

Since we just saw w/ the bug this fixes the change in semantics uncovered a bug so covering both semantics is the right thing to do.